pkg/asset/agent/image/cache.go

```diff
@@ -266,5 +267,16 @@ func DownloadImageFile(baseURL string) (string, error) {
 	}
 	u.location = *parsedURL
 
-	return u.download(imageDataType)
+	for i := 0; i < 3; i++ {
+		path := ""
+		path, err = u.download(imageDataType)
+		if err == nil {
+			return path, nil
+		} else if i < 2 {
```

Comment:
I don't think this check is required. If the download had an error, it would be useful to log it always (at info level)